main.rb

```diff
@@ -0,0 +1,151 @@
+require_relative './student'
+require_relative './teacher'
+require_relative './book'
+require_relative './rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+    @exit = false
+  end
+
+  def run
+    receive_user_input
+  end
+
+  private
+
+  def display_options
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  def receive_user_input
+    loop do
+      display_options
+      print 'Option: '
+      user_input = gets.chomp
+      if user_input.to_i == 7
+        @exit = true
+      else
+        manage_user_input(user_input.to_i)
+      end
+      break if @exit
+    end
+  end
+
+  def manage_user_input(user_input)
+    case user_input
+    when 1
+      display_list_of_books
+    when 2
+      display_list_of_person
+    when 3
+      create_student_or_teacher
+    when 4
+      create_book
+    when 5
+      create_rental
+    when 6
+      display_rentals
+    else
+      puts 'You pressed a wrong option'
+    end
+  end
+
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Has parent permission? [Y/N] '
+    parent_permission = gets.chomp
+    parent_permission = parent_permission.downcase != 'n'
+    @people << Student.new(age.to_i, nil, name: name, parent_permission: parent_permission)
+    puts 'Student created successfully'
+  end
+
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+    print 'Name: '
+    name = gets.chomp
+    print 'Specialization: '
+    specialization = gets.chomp
+    @people << Teacher.new(age.to_i, specialization, name: name)
+    puts 'Teacher created successfully'
+  end
+
+  def create_student_or_teacher
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    option = gets.chomp
+    case option.to_i
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
+    @books << Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number'
+    display_list_of_books(5)
+    book_num = gets.chomp
+    puts 'Select a person from the following list by number (not id)'
+    display_list_of_person(5)
+    person_num = gets.chomp
+    print 'Date: '
+    date = gets.chomp
+    @rentals << Rental.new(date, @books[book_num.to_i], @people[person_num.to_i])
+    puts 'Rental created successfully'
+  end
```

Comment:
- [ ] Creating the rentals is not working because of the date format, please fix the following issue to add the rentals successfully.

![image](https://user-images.githubusercontent.com/55635977/156641665-0416affc-d608-4566-993c-35cbf1f65cd9.png)

Reply:
Hi @Emmyn5600, In that test case in the image, the problem is that you entered no book.

Reply:
This is the test I tried and It's added successfully
![image](https://user-images.githubusercontent.com/24703171/156643547-c6fe6fe3-223f-4957-b787-918ef0d4f378.png)
and this is the test for case 6
![image](https://user-images.githubusercontent.com/24703171/156643709-e22f67d8-e0ca-40ab-8cd5-b824ed73c127.png)